scripts/modules/b2b-account/account-hierarchy.js

```diff
@@ -63,42 +62,46 @@ define(["modules/jquery-mozu", 'modules/api', "underscore", "hyprlive", "modules
         changeParentAccount: function (e) {
             var accountId = e.currentTarget.dataset.mzValue;
             var self = this;
-            inValidParentAccounts = [];
-            var validParentAccounts = [];
-            var currentAccount = this.getAccount(accountId, userChildAccounts);
-            var parentAccount = this.getAccount(currentAccount.account.parentAccountId, self.model.apiModel.data.accounts);
+            var currentAccount = this.model.getAccount(accountId, supportedParentAccounts);
+            var parentAccount = this.model.getAccount(currentAccount.account.parentAccountId, supportedParentAccounts);
+            var filteredParentAccounts = this.getSupportedParentAccounts(currentAccount, parentAccount, supportedParentAccounts);
+            modalView.renderView();
+            modalView.render(currentAccount, parentAccount.account, filteredParentAccounts);
+        },
+        getSupportedParentAccounts: function (currentAccount, parentAccount, supportedParentAccounts) {
+            var inValidParentAccounts = [];
 
             //Exclude current and it's a parent and child accounts
             inValidParentAccounts.push(currentAccount.account);
             if (parentAccount) {
                 inValidParentAccounts.push(parentAccount);
             }
-            this.getInValidParentAccounts(currentAccount.id, userChildAccounts);
-            for (var i = 0; i < userChildAccounts.length; i++) {
-                var invAcc = this.getAccount(userChildAccounts[i].account.id, inValidParentAccounts);
-                if (!invAcc) {
-                    validParentAccounts.push(userChildAccounts[i].account);//Push only valid accounts
-                }
-            }
+            this.getInValidParentAccounts(currentAccount.id, supportedParentAccounts, inValidParentAccounts);
 
-            modalView.renderView();
-            modalView.render(currentAccount, parentAccount, validParentAccounts);
-        },
-        getAccount: function (accountId, accounts) {
-            for (var i = 0; i < accounts.length; i++) {
-                if (accounts[i].id == accountId)
-                    return accounts[i];
+            var filteredParentAccounts = supportedParentAccounts.slice().map(function (ele) {
+                return ele.account;
+            });
+            for (var i = 0; i < inValidParentAccounts.length; i++) {
+                filteredParentAccounts = this.filterAccounts(inValidParentAccounts[i].id, filteredParentAccounts);
             }
+
+            return filteredParentAccounts;
         },
-        getInValidParentAccounts: function (currentAccId, accounts) {
+        getInValidParentAccounts: function (currentAccId, accounts, inValidParentAccounts) {
```

Comment:
Rename `getInValidParentAccounts` to `getInvalidParentAccounts`.

Reply:
done